src/screens/Customers/CustomerProfile.tsx

```diff
@@ -36,14 +37,63 @@ const styles = (theme: Theme) =>
   });
 
 interface CustomerProps extends RouteComponentProps {
+  match: { params: string; isExact: boolean; path: string; url: string; }
   classes: { root: string; content: string; header: string; button: string; buttonSecondary: string };
+  siteName: string;
   location: any;
 }
 
 function CustomerProfile(props: CustomerProps) {
   const { classes, match } = props;
-  const customer: CustomerRecord = props.location.state.customer;
+  const customer = props.location.state.customer;
 
+  // helper functions for getting data
+  const isCurrentReading = (mr: MeterReadingRecord): boolean => {
+    const period: number = parseInt(mr.date.slice(5, 7));
+    const currPer: number = moment().month() + 1;
+    return period == currPer;
+  };
+
+  const getCurrentReading = (customer: CustomerRecord): MeterReadingRecord | undefined => {
+    return customer.meterReadings.find(function (mr: MeterReadingRecord) {
+      return isCurrentReading(mr);
+    });
+  };
+
+  const isStartingMeter = (mr: MeterReadingRecord): boolean => {
+    const period: number = parseInt(mr.date.slice(5, 7)) % 12;
+    const lastPer: number = moment().month();
+    return period == lastPer;
+  };
+
+  const getStartingMeter = (customer: CustomerRecord): MeterReadingRecord | undefined => {
+    return customer.meterReadings.find(function (mr: MeterReadingRecord) {
+      return isStartingMeter(mr);
+    });
+  };
+
+  const getPeriodAndBilled = (customer: CustomerRecord, currReading: MeterReadingRecord, startingMeter: MeterReadingRecord | undefined) => {
```

Comment:
`getPeriodUsageAndAmountBilled` is a more clear name because `Period` standalone is confusing since it means something different. I would also decouple this into a `getPeriodUsage` and `getAmountBilled` since everything else is decoupled